diff --git a/project.clj b/project.clj
--- a/project.clj
+++ b/project.clj
@@ -26,9 +26,10 @@
                  ;; bump versions of various common transitive deps
                  [net.cgrand/parsley "0.9.3" :exclusions [org.clojure/clojure]]
                  [scout "0.1.1"]
-                 [commons-io "2.8.0"]
-                 [commons-codec "1.15"]]
-  :pedantic? :abort
+                 [commons-io "2.17.0"]
+                 [commons-codec "1.15"]
+                 [org.codehaus.plexus/plexus-utils "3.6.1"]]
+  :pedantic? :warn
   ;; checkout-deps don't work with :eval-in :leiningen
   :profiles {:dev {:resource-paths ["leiningen-core/dev-resources"]
                    :test-paths ["leiningen-core/test"]}
