--- a/build.gradle
+++ b/build.gradle
@@ -961,15 +961,15 @@
     implementation libs.slf4jApi
     implementation log4j2Libs
 
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
 
     testImplementation libs.mockitoCore
     testImplementation libs.junitJupiter
     testImplementation testLog4j2Libs
     testImplementation project(':test-common:test-common-internal-api')
     testImplementation project(':test-common:test-common-runtime')
-    testImplementation project(':storage:storage-api').sourceSets.test.output
-    testImplementation project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':storage:storage-api').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
 
     testRuntimeOnly runtimeTestLibs
   }
@@ -1083,14 +1083,14 @@
     implementation libs.slf4jApi
     implementation libs.re2j
 
-    testImplementation project(':clients').sourceSets.test.output
-    testImplementation project(':group-coordinator').sourceSets.test.output
-    testImplementation project(':share-coordinator').sourceSets.test.output
-    testImplementation project(':metadata').sourceSets.test.output
-    testImplementation project(':raft').sourceSets.test.output
-    testImplementation project(':server-common').sourceSets.test.output
-    testImplementation project(':storage:storage-api').sourceSets.test.output
-    testImplementation project(':server').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':group-coordinator').sourceSets.test.output
+    testImplementation rootProject.project(':share-coordinator').sourceSets.test.output
+    testImplementation rootProject.project(':metadata').sourceSets.test.output
+    testImplementation rootProject.project(':raft').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':storage:storage-api').sourceSets.test.output
+    testImplementation rootProject.project(':server').sourceSets.test.output
     testImplementation project(':test-common:test-common-runtime')
     testImplementation project(':test-common:test-common-internal-api')
     testImplementation project(':test-common:test-common-util')
@@ -1361,9 +1361,9 @@
     testImplementation libs.junitJupiter
     testImplementation libs.jqwik
     testImplementation libs.mockitoCore
-    testImplementation project(':clients').sourceSets.test.output
-    testImplementation project(':raft').sourceSets.test.output
-    testImplementation project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':raft').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
     testImplementation project(':test-common:test-common-util')
 
     testRuntimeOnly runtimeTestLibs
@@ -1484,9 +1484,9 @@
     implementation libs.slf4jApi
     implementation libs.hash4j
 
-    testImplementation project(':clients').sourceSets.test.output
-    testImplementation project(':server-common').sourceSets.test.output
-    testImplementation project(':coordinator-common').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':coordinator-common').sourceSets.test.output
     testImplementation libs.jacksonDatabindYaml
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
@@ -1647,7 +1647,7 @@
     testImplementation testLog4j2Libs
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation project(':test-common:test-common-runtime')
     testImplementation project(':test-common:test-common-internal-api')
 
@@ -1711,8 +1711,8 @@
     implementation libs.metrics
     implementation libs.hdrHistogram
 
-    testImplementation project(':clients').sourceSets.test.output
-    testImplementation project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
     testImplementation testLog4j2Libs
@@ -1747,9 +1747,9 @@
     implementation libs.metrics
     implementation libs.slf4jApi
 
-    testImplementation project(':clients').sourceSets.test.output
-    testImplementation project(':server-common').sourceSets.test.output
-    testImplementation project(':coordinator-common').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':coordinator-common').sourceSets.test.output
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
     testImplementation testLog4j2Libs
@@ -2042,10 +2042,10 @@
     testImplementation project(':metadata')
     testImplementation project(':server')
     testImplementation project(':storage')
-    testImplementation project(':core').sourceSets.test.output
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':core').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     implementation project(':server-common')
-    testImplementation project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
     testImplementation project(':metadata')
     implementation project(':group-coordinator')
     implementation project(':group-coordinator:group-coordinator-api')
@@ -2080,9 +2080,9 @@
     implementation libs.slf4jApi
 
     testImplementation project(':server-common')
-    testImplementation project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
     testImplementation project(':clients')
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation libs.jacksonDatabindYaml
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
@@ -2179,7 +2179,7 @@
     implementation libs.slf4jApi
 
     testImplementation project(':clients')
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation libs.jacksonDatabindYaml
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
@@ -2239,7 +2239,7 @@
     implementation libs.slf4jApi
 
     testImplementation project(':clients')
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
     testImplementation testLog4j2Libs
@@ -2309,16 +2309,16 @@
     implementation libs.metrics
 
     testImplementation project(':clients')
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation project(':core')
-    testImplementation project(':core').sourceSets.test.output
-    testImplementation project(':storage:storage-api').sourceSets.test.output
+    testImplementation rootProject.project(':core').sourceSets.test.output
+    testImplementation rootProject.project(':storage:storage-api').sourceSets.test.output
     testImplementation project(':test-common:test-common-internal-api')
     testImplementation project(':test-common:test-common-runtime')
     testImplementation project(':test-common:test-common-util')
     testImplementation project(':server')
     testImplementation project(':server-common')
-    testImplementation project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
     testImplementation project(':transaction-coordinator')
     testImplementation libs.hamcrest
     testImplementation libs.jacksonDatabindYaml
@@ -2501,23 +2501,23 @@
     compileOnly libs.spotbugs
 
     testImplementation project(':clients')
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation project(':server')
-    testImplementation project(':server').sourceSets.test.output
+    testImplementation rootProject.project(':server').sourceSets.test.output
     testImplementation project(':core')
-    testImplementation project(':core').sourceSets.test.output
+    testImplementation rootProject.project(':core').sourceSets.test.output
     testImplementation project(':test-common:test-common-internal-api')
     testImplementation project(':test-common:test-common-runtime')
     testImplementation project(':server-common')
-    testImplementation project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
     testImplementation project(':connect:api')
     testImplementation project(':connect:runtime')
-    testImplementation project(':connect:runtime').sourceSets.test.output
-    testImplementation project(':storage:storage-api').sourceSets.main.output
-    testImplementation project(':storage').sourceSets.test.output
+    testImplementation rootProject.project(':connect:runtime').sourceSets.test.output
+    testImplementation rootProject.project(':storage:storage-api').sourceSets.main.output
+    testImplementation rootProject.project(':storage').sourceSets.test.output
     testImplementation project(':streams')
-    testImplementation project(':streams').sourceSets.test.output
-    testImplementation project(':streams:integration-tests').sourceSets.test.output
+    testImplementation rootProject.project(':streams').sourceSets.test.output
+    testImplementation rootProject.project(':streams:integration-tests').sourceSets.test.output
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
     testImplementation libs.mockitoJunitJupiter // supports MockitoExtension
@@ -2586,7 +2586,7 @@
     implementation project(':group-coordinator:group-coordinator-api')
 
     testImplementation project(':clients')
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation project(':group-coordinator')
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
@@ -2634,10 +2634,10 @@
     implementation libs.jacksonJakartarsJsonProvider
 
     testImplementation project(':clients')
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation project(':core')
     testImplementation project(':server-common')
-    testImplementation project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
     testImplementation libs.jacksonDatabindYaml
     testImplementation libs.junitJupiter
     testImplementation testLog4j2Libs
@@ -2686,7 +2686,7 @@
     testCompileOnly project(':streams:test-utils')
     testCompileOnly libs.bndlib
 
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation libs.jacksonDatabindYaml
     testImplementation libs.junitJupiter
     testImplementation libs.bcpkix
@@ -2828,8 +2828,8 @@
     api project(':streams')
 
     api libs.scalaLibrary
-    testImplementation project(':streams').sourceSets.test.output
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':streams').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation project(':streams:test-utils')
 
     testImplementation libs.jacksonDatabindYaml
@@ -2881,13 +2881,13 @@
     implementation libs.slf4jApi
     implementation libs.scalaLibrary
 
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation project(':group-coordinator')
     testImplementation project(':server')
     testImplementation project(':server-common')
-    testImplementation project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
     testImplementation project(':storage')
-    testImplementation project(':streams').sourceSets.test.output
+    testImplementation rootProject.project(':streams').sourceSets.test.output
     testImplementation project(':streams:streams-scala')
     testImplementation project(':test-common:test-common-runtime')
     testImplementation project(':tools')
@@ -2935,7 +2935,7 @@
 
     implementation libs.slf4jApi
 
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation libs.jacksonDatabindYaml
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
@@ -2971,7 +2971,7 @@
     implementation libs.jacksonAnnotations
 
     testImplementation project(':streams:test-utils')
-    testImplementation project(':clients').sourceSets.test.output // for org.apache.kafka.test.IntegrationTest
+    testImplementation rootProject.project(':clients').sourceSets.test.output // for org.apache.kafka.test.IntegrationTest
     testImplementation libs.junitJupiter
     testImplementation libs.hamcrest
     testImplementation testLog4j2Libs
@@ -3398,7 +3398,7 @@
     implementation project(':raft')
     implementation project(':clients')
     implementation project(':coordinator-common')
-    implementation project(':coordinator-common').sourceSets.test.output
+    implementation rootProject.project(':coordinator-common').sourceSets.test.output
     implementation project(':group-coordinator')
     implementation project(':group-coordinator:group-coordinator-api')
     implementation project(':metadata')
@@ -3409,10 +3409,10 @@
     implementation project(':connect:api')
     implementation project(':connect:transforms')
     implementation project(':connect:json')
-    implementation project(':clients').sourceSets.test.output
-    implementation project(':core').sourceSets.test.output
-    implementation project(':server-common').sourceSets.test.output
-    implementation project(':metadata').sourceSets.test.output
+    implementation rootProject.project(':clients').sourceSets.test.output
+    implementation rootProject.project(':core').sourceSets.test.output
+    implementation rootProject.project(':server-common').sourceSets.test.output
+    implementation rootProject.project(':metadata').sourceSets.test.output
 
     implementation libs.jmhCore
     annotationProcessor libs.jmhGeneratorAnnProcess
@@ -3467,7 +3467,7 @@
     implementation libs.slf4jApi
 
     testImplementation libs.junitJupiter
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation testLog4j2Libs
 
     testRuntimeOnly runtimeTestLibs
@@ -3502,7 +3502,7 @@
     implementation libs.slf4jApi
 
     testImplementation libs.junitJupiter
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation testLog4j2Libs
 
     testRuntimeOnly runtimeTestLibs
@@ -3542,7 +3542,7 @@
 
     testImplementation libs.junitJupiter
     testImplementation testLog4j2Libs
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
 
     testRuntimeOnly runtimeTestLibs
   }
@@ -3616,7 +3616,7 @@
     swagger libs.jakartaServletApi
     swagger libs.jaxrs2Jakarta
 
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation project(':core')
     testImplementation project(':server')
     testImplementation project(':metadata')
@@ -3629,7 +3629,7 @@
     testImplementation project(':group-coordinator')
     testImplementation project(':storage')
     testImplementation project(':connect:test-plugins')
-    testImplementation project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
 
     testImplementation libs.jacksonDatabindYaml
     testImplementation libs.junitJupiter
@@ -3743,12 +3743,12 @@
     testImplementation libs.junitJupiter
     testImplementation libs.mockitoCore
 
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
     testImplementation project(':connect:runtime')
-    testImplementation project(':connect:runtime').sourceSets.test.output
+    testImplementation rootProject.project(':connect:runtime').sourceSets.test.output
     testImplementation project(':core')
     testImplementation project(':test-common:test-common-runtime')
-    testImplementation project(':server-common').sourceSets.test.output
+    testImplementation rootProject.project(':server-common').sourceSets.test.output
 
     testRuntimeOnly runtimeTestLibs
   }
@@ -3787,7 +3787,7 @@
     testImplementation libs.mockitoCore
     testImplementation libs.junitJupiter
     testImplementation testLog4j2Libs
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
 
     testRuntimeOnly libs.jerseyContainerServlet
     testRuntimeOnly runtimeTestLibs
@@ -3848,8 +3848,8 @@
     testImplementation libs.junitJupiter
     testImplementation libs.bndlib
     testImplementation libs.mockitoCore
-    testImplementation project(':clients').sourceSets.test.output
-    testImplementation project(':connect:runtime').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':connect:runtime').sourceSets.test.output
     testImplementation project(':core')
     testImplementation project(':test-common:test-common-runtime')
     testImplementation project(':server')
@@ -3918,7 +3918,7 @@
 
     testImplementation testLog4j2Libs
     testImplementation libs.junitJupiter
-    testImplementation project(':clients').sourceSets.test.output
+    testImplementation rootProject.project(':clients').sourceSets.test.output
 
     testRuntimeOnly runtimeTestLibs
   }
