diff --git a/LICENSE-binary b/LICENSE-binary
index b601ccebfb..f375a6445a 100644
--- a/LICENSE-binary
+++ b/LICENSE-binary
@@ -238,10 +238,10 @@ License Version 2.0:
 - jetty-util-12.0.22
 - jose4j-0.9.6
 - jspecify-1.0.0
-- log4j-api-2.24.3
-- log4j-core-2.24.3
-- log4j-slf4j-impl-2.24.3
-- log4j-1.2-api-2.24.3
+- log4j-api-2.25.3
+- log4j-core-2.25.3
+- log4j-slf4j-impl-2.25.3
+- log4j-1.2-api-2.25.3
 - lz4-java-1.8.0
 - maven-artifact-3.9.6
 - metrics-core-2.2.0
diff --git a/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Loggers.java b/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Loggers.java
index 3767e31ac7..1a79698ae9 100644
--- a/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Loggers.java
+++ b/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Loggers.java
@@ -255,7 +255,7 @@ public abstract class Loggers {
             LoggerContext context = (LoggerContext) LogManager.getContext(false);
             var results = new HashMap<String, org.apache.logging.log4j.core.Logger>();
             context.getConfiguration().getLoggers().forEach((name, logger) -> results.put(name, loggerContext.getLogger(name)));
-            context.getLoggerRegistry().getLoggers().forEach(logger -> results.put(logger.getName(), logger));
+            context.getLoggers().forEach(logger -> results.put(logger.getName(), logger));
             return results;
         }
 
diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index 10173814e5..4006493c9c 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -105,7 +105,7 @@ versions += [
   kafka_38: "3.8.1",
   kafka_39: "3.9.1",
   kafka_40: "4.0.0",
-  log4j2: "2.24.3",
+  log4j2: "2.25.3",
   // When updating lz4 make sure the compression levels in org.apache.kafka.common.record.CompressionType are still valid
   lz4: "1.10.1",
   mavenArtifact: "3.9.6",
