diff --git a/vendor/knative.dev/pkg/observability/semconv/semconv.go b/vendor/knative.dev/pkg/observability/semconv/semconv.go
--- a/vendor/knative.dev/pkg/observability/semconv/semconv.go
+++ b/vendor/knative.dev/pkg/observability/semconv/semconv.go
@@ -18,7 +18,7 @@
 
 import (
 	"go.opentelemetry.io/otel/attribute"
-	semconv "go.opentelemetry.io/otel/semconv/v1.40.0"
+	semconv "go.opentelemetry.io/otel/semconv/v1.41.0"
 )
 
 const (
