diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index 155d6c074..943b617ff 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -121,7 +121,7 @@ opentelemetry-bom-alpha = { module = "io.opentelemetry:opentelemetry-bom-alpha",
 opentelemetry-instrumentation-bom-alpha = { module = "io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom-alpha", version.ref = "opentelemetryAlpha" }
 picocli = { module = "info.picocli:picocli-codegen", version.ref = "picocli" }
 picocli-codegen = { module = "info.picocli:picocli-codegen", version.ref = "picocli" }
-postgresql = { module = "org.postgresql:postgresql", version = "42.7.10" }
+postgresql = { module = "org.postgresql:postgresql", version = "42.7.11" }
 protobuf-java = { module = "com.google.protobuf:protobuf-java", version.ref = "protobuf" }
 quarkus-azure-services-bom = { module = "io.quarkiverse.azureservices:quarkus-azure-services-bom", version = "1.2.2" }
 quarkus-bom = { module = "io.quarkus.platform:quarkus-bom", version.ref = "quarkus" }
diff --git a/servers/quarkus-server/build.gradle.kts b/servers/quarkus-server/build.gradle.kts
index 8fa5abd3a..655c9f876 100644
--- a/servers/quarkus-server/build.gradle.kts
+++ b/servers/quarkus-server/build.gradle.kts
@@ -85,6 +85,14 @@ configurations.configureEach {
   }
 }
 
+configurations.configureEach {
+  resolutionStrategy {
+    force(
+      "org.postgresql:postgresql:42.7.11",
+    )
+  }
+}
+
 val quarkusRunner by
   configurations.creating {
     description = "Used to reference the generated runner-jar (either fast-jar or uber-jar)"
