diff --git a/gradle/libs.versions.toml b/gradle/libs.versions.toml
index 16e7bc4ec..155d6c074 100644
--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -25,7 +25,7 @@ opentelemetry = "1.61.0"
 opentelemetryAlpha = "1.31.0-alpha"
 picocli = "4.7.6"
 protobuf = "4.34.1"
-quarkus = "3.34.5"
+quarkus = "3.35.2"
 quarkusVault = "4.7.0"
 slf4j = "2.0.17"
 undertow = "2.3.24.Final"
diff --git a/servers/quarkus-server/build.gradle.kts b/servers/quarkus-server/build.gradle.kts
index 538479a57..8fa5abd3a 100644
--- a/servers/quarkus-server/build.gradle.kts
+++ b/servers/quarkus-server/build.gradle.kts
@@ -77,6 +77,14 @@ configurations.configureEach {
   }
 }
 
+configurations.configureEach {
+  resolutionStrategy {
+    force(
+      "io.quarkus:quarkus-vertx-http:3.35.2",
+    )
+  }
+}
+
 val quarkusRunner by
   configurations.creating {
     description = "Used to reference the generated runner-jar (either fast-jar or uber-jar)"
